### PR TITLE
JAPI-204 Support query limit in DFSCLIENT

### PR DESCRIPTION
**Pull request by jpmcmu**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [X] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java [hunk 1]
```diff
@@ -47,10 +47,33 @@ public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilde
...
+     * @param originalRD
+     *            the record defintion for the dataset
+     * @param recBuilder
+     *            the IRecordBuilder used to construct records
+     * @param projectedRD
+     *            the requested record format
+     * @param connectTimeout 
+     *            the connectiont timeout in seconds
+     * @param limit 
+     *            the record limit for reading. -1 specifies no limit
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java]: If we can, let's be as descriptive here with the behavior of the limit... ie. first n recs from 1st file part, etc.

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java [hunk 2]
```diff
@@ -47,10 +47,33 @@ public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilde
...
      * @param recBuilder
      *            the IRecordBuilder used to construct records
+     * @param connectTimeout 
+     *            the connectiont timeout in seconds
      * @param projectedRD
      *            the requested record format
      */
     public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilder recBuilder, int connectTimeout) throws Exception
+    {
+        this(dp,originalRD,recBuilder,connectTimeout,-1);
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java]: Can we create a Symbol to signify no limit, that way we can change the actual value behind the scenes if we need to

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 3]
```diff
@@ -94,10 +114,10 @@ public RowServiceInputStream(DataPartition dp, FieldDef rd, FieldDef pRd, int co
         int copycount = dataPart.getCopyCount();
         for (int index = 0; index < copycount; index++)
         {
-            String currentCopyLocation = dataPart.getCopyIP(index);
-            if(Network.isLocalAddress(currentCopyLocation))
-                prioritizedCopyIndexes.add(0,index);
-            else
+            // String currentCopyLocation = dataPart.getCopyIP(index);
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java]: let's keep this code in place until we decide to explicitly remove it

File: wsclient/pom.xml [hunk 4]
```diff
@@ -3,7 +3,7 @@
     <groupId>org.hpccsystems</groupId>
     <artifactId>wsclient</artifactId>
     <packaging>jar</packaging>
-    <version>7.2.16-SNAPSHOT</version>
+    <version>7.2.16</version>
```
Comment (rpastrana) [inline wsclient/pom.xml]: let's keep this as snapshot

**Comment by jpmcmu** [comment]
@rpastrana code review changes have been pushed up. Will squash and rebase from master if it looks good

**Comment by jpmcmu** [comment]
@rpastrana rebased

**Pull request merged**